Add the workflow to deploy to Heroku and release PR template (#101)
nimblehq/elixir-templates: priv/templates/nimble_template/.github/workflows/README.md

## Deploy to Heroku Workflow

### Requirements
- A pre-generated [Heroku App](https://devcenter.heroku.com/articles/creating-apps)
- A Heroku API key. It can be generated under your [Account Settings](https://dashboard.heroku.com/account#api-key)
- Three Heroku config vars:  
  - **DATABASE_URL**: It will be created automatically when the [PostgreSQL add-on](https://elements.heroku.com/addons/heroku-postgresql) is added.
  - **HOST_URL**: if your app name is `acme`, the value of this var is: `acme.herokuapp.com`
  - **SECRET_KEY_BASE**: use the `mix phx.gen.secret` to get a new secret and set it as the value of this var.

### How to use
- Defining two [Github secrets](https://docs.github.com/en/actions/reference/encrypted-secrets) to hold the value of Heroku API key and Heroku app name:
  - HEROKU_API_KEY
  - HEROKU_APP_NAME
- Making some configuration changes to make the app ready for Heroku (read more in the [official documentation](https://hexdocs.pm/phoenix/heroku.html#making-our-project-ready-for-heroku)) that: 
  tell Phoenix to use Heroku URL and enforce the SSL usage. Also, bind to the port requested by Heroku in the `$PORT` environment variable.

>   Find the url line in your `config/prod.exs`: 

Author: Changed in 252f3a1.

  ```elixir
  url: [host: "example.com", port: 80],
  ```
  and replace it with this:
  ```elixir
  http: [port: {:system, "PORT"}],
  url: [scheme: "https", host: System.get_env("HOST_URL"), port: 443],
  force_ssl: [rewrite_on: [:x_forwarded_proto]],
  ```

>   Then open up the `config/prod.secret.exs` and uncomment the `# ssl: true`

Author: also changed in 252f3a1.

  ```elixir
  config :gscraper_web, GscraperWeb.Repo,
    ssl: true,
    ...  
  ```

>   if you plan on using websockets, then we will need to decrease the timeout for the websocket transport in `lib/hello_web/endpoint.ex`. 

Reviewer: ```suggestion
  If you plan on using WebSocket, the timeout for the WebSocket transport needs to be decreased in `lib/hello_web/endpoint.ex`. 
```

Author: also changed in in 252f3a1.

# Github Actions

The following workflows are supported. Depends on what you choose when generating the project, some or all of them would be available:

- Test
- [Deploy to Heroku](#deploy-to-heroku-workflow)

Please refer to the following sections for usage instruction.

> ## Deploy to Heroku Workflow

Reviewer 2: Should we only generate this content when the user chooses `Deploy to Heroku` option, because if the user doesn't choose it, it doesn't make sense to have this section in the README file

Author: <img width="830" alt="Screen Shot 2021-12-24 at 13 32 43" src="https://user-images.githubusercontent.com/1896814/147325736-7741fd65-49b6-4cc3-b421-efc2fb3dc0fe.png">

do you have any ideas to prepend the content to 2 places when a new workflow option is selected?

e.g. if the user selects to generate the Wiki workflow, we also need to append to the above 2 places.
As a workaround, I added a sentence to the README file

> The following workflows are supported. Depends on what you choose when generating the project, some or all of them would be available:

Reviewer 2: We can put the condition into this README.md file @longnd I will try to adjust that on a separate PR
